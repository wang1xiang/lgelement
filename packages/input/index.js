import WangInput from './src/input.vue'

WangInput.install = Vue => {
  Vue.component(WangInput.name, WangInput)
}

export default WangInput
