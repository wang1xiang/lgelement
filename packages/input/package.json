{
  "name": "wang-input",
  "version": "0.0.0",
  "description": "wang-input component",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "author": "xiangzi <7555638369@qq.com>",
  "license": "MIT",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "lodash": "3"
  },
  "keywords": []
}
