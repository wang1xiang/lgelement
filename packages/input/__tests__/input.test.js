import input from '../src/input.vue'
import { mount } from '@vue/test-utils'

// 创建代码块 将input相关测试都添加到这里
describe('wang-input', () => {
  test('input-text', () => {
    // 挂载组件 只是内存中的挂载 返回一个包裹器
    const wrapper = mount(input)
    expect(wrapper.html()).toContain('input type="text"')
  })

  test('input-password', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
      },
    })
    expect(wrapper.html()).toContain('input type="password"')
  })

  test('input-password', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: 'admin',
      },
    })
    expect(wrapper.props('value')).toBe('admin')
  })

  // 快照
  test('input-snapshot', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: 'admin',
      },
    })
    // 快照 第一次运行会将wrapper.vm.$el的内容存储在./__snapshots__/input.test.js.snap中
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
