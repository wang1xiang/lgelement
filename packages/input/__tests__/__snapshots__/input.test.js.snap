// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wang-input input-snapshot 1`] = `
<div>
  <input
    type="password"
  />
</div>
`;
